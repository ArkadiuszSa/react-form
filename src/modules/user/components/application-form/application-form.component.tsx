import * as React from "react";
import "react-datepicker/dist/react-datepicker.css";
import DatePicker from "react-datepicker";
import Button from "@material-ui/core/Button";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import FormHelperText from "@material-ui/core/FormHelperText";
import FormControl from "@material-ui/core/FormControl";
import MaskedInput from "react-text-mask";

import "./application-form.scss";
import {
  fetchHappening,
  handleFormChange,
  handleDateChange,
  submitApplicationForm,
  resetApplicationForm
} from "./application-form.actions";

export interface FormProps {
  match: any;
  history: any;
  fetchHappening: Function;
  resetApplicationForm: Function;
  handleFormChange: Function;
  handleDateChange: any;
  submitApplicationForm: Function;

  title: string;
  firstName: string;
  firstNameErr: string;
  lastName: string;
  lastNameErr: string;
  email: string;
  emailErr: string;
  date: string;
  dateErr: string;
  avaibleDates: any[];
  selectedDate: any;
  submitInfo: any;
}

class ApplicationForm extends React.Component<FormProps, {}> {
  public happeningId = "";
  public disabledInput: boolean = false;
  constructor(props) {
    super(props);

    this.happeningId = this.props.match.params.id;
  }
  componentWillMount() {
    this.props.resetApplicationForm();
    this.disabledInput = false;
  }

  componentWillUnmount() {
    this.props.resetApplicationForm();
  }

  componentDidMount() {
    this.props.fetchHappening(this.happeningId);
  }

  TextMaskCustom(props) {
    const { inputRef, ...other } = props;

    return (
      <MaskedInput
        {...other}
        ref={inputRef}
        mask={[/\d/, /\d/, /\d/, /\d/, "-", /\d/, /\d/, "-", /\d/, /\d/]}
        keepCharPositions={true}
        placeholder={"yyyy-mm-dd"}
      />
    );
  }

  render() {
    if (this.props.submitInfo === "Form correctly saved!") {
      this.disabledInput = true;
      setTimeout(() => {
        this.props.history.push("/");
      }, 2000);
    }
    return (
      <div className="form-container">
        <form
          className="form"
          onSubmit={e =>
            this.props.submitApplicationForm(e, {
              happeningId: this.happeningId,
              firstName: this.props.firstName,
              lastName: this.props.lastName,
              email: this.props.email,
              date: this.props.date,
              avaibleDates: this.props.avaibleDates
            })
          }
        >
          <p className="title">Sign up for {this.props.title}</p>

          <FormControl className="text-field">
            <InputLabel>First name</InputLabel>
            <Input
              disabled={this.disabledInput}
              name="firstName"
              value={this.props.firstName}
              onChange={e => this.props.handleFormChange(e)}
            />
            <FormHelperText className="field-error">
              {this.props.firstNameErr}
            </FormHelperText>
          </FormControl>

          <FormControl className="text-field">
            <InputLabel>Last name</InputLabel>
            <Input
              disabled={this.disabledInput}
              name="lastName"
              value={this.props.lastName}
              onChange={e => this.props.handleFormChange(e)}
            />
            <FormHelperText className="field-error">
              {this.props.lastNameErr}
            </FormHelperText>
          </FormControl>

          <FormControl className="text-field">
            <InputLabel>Email</InputLabel>
            <Input
              disabled={this.disabledInput}
              name="email"
              value={this.props.email}
              onChange={e => this.props.handleFormChange(e)}
            />
            <FormHelperText className="field-error">
              {this.props.emailErr}
            </FormHelperText>
          </FormControl>

          <DatePicker
            selected={this.props.selectedDate}
            onChange={this.props.handleDateChange}
            includeDates={this.props.avaibleDates}
            customInput={
              <FormControl className="text-field">
                <InputLabel>Date</InputLabel>
                <Input
                  disabled={this.disabledInput}
                  name="date"
                  value={this.props.date}
                  onChange={e => this.props.handleFormChange(e)}
                  inputComponent={this.TextMaskCustom}
                />
                <FormHelperText className="field-error">
                  {this.props.dateErr}
                </FormHelperText>
              </FormControl>
            }
          />

          <p className="submit-info">{this.props.submitInfo}</p>
          <Button variant="outlined" type="submit" className="submit-button">
            Submit
          </Button>
        </form>
      </div>
    );
  }
}

export default ApplicationForm;
