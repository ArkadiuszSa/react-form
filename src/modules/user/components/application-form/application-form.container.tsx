import { connect } from "react-redux";
import {
  fetchHappening,
  handleFormChange,
  handleDateChange,
  validationApplicationFormFailure,
  submitApplicationFormSucces,
  submitApplicationFormFailure,
  resetApplicationFormSubmitInfo,
  validationApplicationFormSucces,
  resetApplicationForm
} from "./application-form.actions";
import ApplicationFormService from "./application-form.service";
import ApplicationForm from "./application-form.component";

const applicationFormService = new ApplicationFormService();

function mapStateToProps(state, ownProps) {
  return {
    title: state.applicationForm.title,
    avaibleDates: state.applicationForm.avaibleDates,
    selectedDate: state.applicationForm.selectedDate,
    firstName: state.applicationForm.firstName,
    firstNameErr: state.applicationForm.firstNameErr,
    lastName: state.applicationForm.lastName,
    lastNameErr: state.applicationForm.lastNameErr,
    email: state.applicationForm.email,
    emailErr: state.applicationForm.emailErr,
    date: state.applicationForm.date,
    dateErr: state.applicationForm.dateErr,
    submitInfo: state.applicationForm.submitInfo
  };
}

const mapDispatchToProps = dispatch => {
  return {
    fetchHappening: id => {
      applicationFormService.getHappening(id).then(response => {
        dispatch(fetchHappening(response));
      });
    },

    resetApplicationForm: () => {
      dispatch(resetApplicationForm());
    },

    handleFormChange: (e: React.FormEvent<EventTarget>) => {
      let target = e.target as HTMLInputElement;
      dispatch(
        handleFormChange({
          name: target.name,
          value: target.value
        })
      );
      dispatch(resetApplicationFormSubmitInfo());
    },

    handleDateChange: date => {
      let newDate = date.format("YYYY-MM-DD");
      dispatch(
        handleDateChange({
          date: newDate,
          selectedDate: date
        })
      );
      dispatch(resetApplicationFormSubmitInfo());
    },

    submitApplicationForm: (e, application) => {
      e.preventDefault();
      const validationRes = applicationFormService.validateForm(application);

      if (validationRes.isError) {
        dispatch(
          validationApplicationFormFailure({
            ...validationRes.errors
          })
        );
      } else {
        dispatch(validationApplicationFormSucces());

        applicationFormService
          .AddNewApplication(application)
          .then(response => {
            dispatch(submitApplicationFormSucces());
          })
          .catch(error => {
            if (error.response.data.errors[0].param === "email") {
              dispatch(
                submitApplicationFormFailure("The provided email is invalid!")
              );
            } else if (
              error.response.data.errors[0] ===
              "This email is already signed up for this event"
            ) {
              dispatch(
                submitApplicationFormFailure(
                  "This email is already signed up for this event!"
                )
              );
            } else {
              dispatch(
                submitApplicationFormFailure(
                  "Form has not been saved correctly! Please check the connection."
                )
              );
            }
          });
      }
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ApplicationForm);
