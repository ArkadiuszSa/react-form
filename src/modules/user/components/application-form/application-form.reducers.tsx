import {
  FETCH_APPLICATION_FORM_HAPPENING,
  RESET_APPLICATION_FORM,
  SUBMIT_APPLICATION_FORM,
  HANDLE_FORM_CHANGE,
  HANDLE_DATE_CHANGE,
  VALIDATION_APPLICATION_FORM_FAILURE,
  SUBMIT_APPLICATION_FORM_FAILURE,
  RESET_APPLICATION_FORM_SUBMIT_INFO,
  SUBMIT_APPLICATION_FORM_SUCCES,
  VALIDATION_APPLICATION_FORM_SUCCES
} from "./application-form.actions";

const INITIAL_STATE = {
  title: "",
  firstName: "",
  firstNameErr: "",
  lastName: "",
  lastNameErr: "",
  email: "",
  emailErr: "",
  date: "",
  dateErr: "",
  submitInfo: "",
  avaibleDates: [],
  selectedDate: ""
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case FETCH_APPLICATION_FORM_HAPPENING:
      return {
        ...state,
        title: action.payload.title,
        avaibleDates: action.payload.avaibleDates,
        selectedDate: action.payload.selectedDate
      };

    case RESET_APPLICATION_FORM:
      return {
        ...state,
        ...INITIAL_STATE
      };

    case HANDLE_FORM_CHANGE:
      return {
        ...state,
        [action.payload.name]: action.payload.value
      };
    case HANDLE_DATE_CHANGE:
      return {
        ...state,
        date: action.payload.date,
        selectedDate: action.payload.selectedDate
      };
    case VALIDATION_APPLICATION_FORM_FAILURE:
      return {
        ...state,
        ...action.payload
      };
    case VALIDATION_APPLICATION_FORM_SUCCES:
      return {
        ...state,
        firstNameErr: "",
        lastNameErr: "",
        emailErr: "",
        dateErr: ""
      };
    case SUBMIT_APPLICATION_FORM_SUCCES:
      return {
        ...state,
        submitInfo: action.payload
      };
    case SUBMIT_APPLICATION_FORM_FAILURE:
      return {
        ...state,
        submitInfo: action.payload
      };
    case RESET_APPLICATION_FORM_SUBMIT_INFO:
      return {
        ...state,
        submitInfo: action.payload
      };

    default:
      return state;
  }
};
