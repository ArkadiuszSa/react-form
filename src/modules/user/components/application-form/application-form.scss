$primary-color: #0083ff;
.form-container {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;

  .form {
    width: 60%;
    max-width: 900px;
    padding: 50px;
    display: flex;
    flex-direction: column;
    background: white;
    -webkit-box-shadow: 1px 5px 15px 5px rgba(0, 0, 0, 0.5);
    box-shadow: 1px 5px 15px 5px rgba(0, 0, 0, 0.5);

    .text-field {
      border-color: $primary-color;
      outline-color: #0083ff;
      margin-top: 0;
    }

    .title {
      text-align: center;
      font-size: 1.8em;
      margin-top: 0;
    }

    .submit-button {
      margin-top: 12px;
      color: $primary-color;
      border-color: $primary-color;
    }

    .submit-button:hover {
      background: rgba(0, 131, 255, 0.1);
    }

    .field-error {
      color: red;
    }

    .submit-info {
      color: $primary-color;
    }
  }
}

.react-datepicker-wrapper {
  width: 100%;
}

.react-datepicker-wrapper {
  div {
    width: 100%;
  }
}
