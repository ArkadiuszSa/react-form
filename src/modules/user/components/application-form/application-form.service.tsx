import axios from 'axios';
import * as moment from 'moment';

export interface HappeningData {
  data: {
    title: string,
    days: any,
  }
}

export default class ApplicationFormService {
  constructor() {

  }

  public getHappening(id) {
    return axios.get("http://localhost:4000/api/happening/" + id)
      .then((response: HappeningData) => {

        let dates = [];
        for (let date of response.data.days) {
          dates.push(moment(date))
        }

        return {
          title: response.data.title,
          avaibleDates: dates
        }
      })
  }

  public AddNewApplication(application) {
    return axios.post("http://localhost:4000/api/application", application)
      .then((response) => {
        return('Form correctly saved!')
      })
      .catch((error) => {
        return ('Form has not been saved correctly!' )
      });
  }

  public validateForm(stateValues) {
    let isError = false;
    const errors = {
      firstNameErr: "",
      lastNameErr: "",
      emailErr: "",
      dateErr: ""
    };

    let invalidDate = true;
    for (let date of stateValues.avaibleDates) {
      if (stateValues.date === date.format("YYYY-MM-DD")) {
        invalidDate = false;
      }
    }
    if (invalidDate) {
      isError = true;
      errors.dateErr = "Date must match to event dates";
    }

    let emailRegex = /\S+@\S+\.\S+/;
    if (!emailRegex.test(stateValues.email)) {
      isError = true;
      errors.emailErr = "Requires valid email";
    }

    if (stateValues.firstName === "") {
      isError = true;
      errors.firstNameErr = "Requires first name";
    }
    if (stateValues.lastName === "") {
      isError = true;
      errors.lastNameErr = "Requires last name";
    }
    if (stateValues.date === "") {
      isError = true;
      errors.dateErr = "Requires date";
    }

    return { isError: isError, errors: errors };
  }


}
