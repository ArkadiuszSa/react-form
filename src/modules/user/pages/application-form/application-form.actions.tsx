export const FETCH_APPLICATION_HAPPENING_SUCCES =
  "FETCH_APPLICATION_HAPPENING_SUCCES";
export const FETCH_APPLICATION_HAPPENING_FAILURE =
  "FETCH_APPLICATION_HAPPENING_FAILURE";
export const RESET_APPLICATION_FORM = "RESET_APPLICATION_FORM";

export const HANDLE_FORM_CHANGE = "HANDLE_FORM_CHANGE";
export const HANDLE_DATE_CHANGE = "HANDLE_DATE_CHANGE";

export const VALIDATION_APPLICATION_FORM_FAILURE =
  "VALIDATION_APPLICATION_FORM_FAILURE";
export const VALIDATION_APPLICATION_FORM_SUCCES =
  "VALIDATION_APPLICATION_FORM_SUCCES";

export const SUBMIT_APPLICATION_FORM = "SUBMIT_APPLICATION_FORM";
export const SUBMIT_APPLICATION_FORM_SUCCES = "SUBMIT_APPLICATION_FORM_SUCCES";
export const SUBMIT_APPLICATION_FORM_FAILURE =
  "SUBMIT_APPLICATION_FORM_FAILURE";

export const RESET_APPLICATION_FORM_SUBMIT_INFO =
  "RESET_APPLICATION_FORM_SUBMIT_INFO";

export function fetchHappeningSucces(data) {
  return {
    type: FETCH_APPLICATION_HAPPENING_SUCCES,
    payload: data
  };
}

export function fetchHappeningFailure() {
  return {
    type: FETCH_APPLICATION_HAPPENING_FAILURE,
    payload: "There is problem with server connection"
  };
}

export function resetApplicationForm() {
  return {
    type: RESET_APPLICATION_FORM
  };
}

export function handleFormChange(formElement) {
  return {
    type: HANDLE_FORM_CHANGE,
    payload: formElement
  };
}

export function handleDateChange(dates) {
  return {
    type: HANDLE_DATE_CHANGE,
    payload: dates
  };
}

export function validationApplicationFormFailure(errors) {
  return {
    type: VALIDATION_APPLICATION_FORM_FAILURE,
    payload: errors
  };
}

export function validationApplicationFormSucces() {
  return {
    type: VALIDATION_APPLICATION_FORM_SUCCES
  };
}

export function submitApplicationForm(data) {
  return {
    type: SUBMIT_APPLICATION_FORM,
    payload: data
  };
}

export function submitApplicationFormSucces() {
  return {
    type: SUBMIT_APPLICATION_FORM_SUCCES,
    payload: "Form correctly saved!"
  };
}

export function submitApplicationFormFailure(error) {
  return {
    type: SUBMIT_APPLICATION_FORM_FAILURE,
    payload: error
  };
}

export function resetApplicationFormSubmitInfo() {
  return {
    type: RESET_APPLICATION_FORM_SUBMIT_INFO
  };
}
