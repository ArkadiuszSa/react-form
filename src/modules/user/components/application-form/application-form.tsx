import * as React from "react";
import * as moment from 'moment';
import axios from 'axios';

import "./application-form.scss";
import ApplicationFormService from "./application-form.service"

import 'react-datepicker/dist/react-datepicker.css';
import DatePicker from 'react-datepicker';

import Button from '@material-ui/core/Button';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import MaskedInput from 'react-text-mask';

export interface FormState {
  title?: string,
  firstName?: string,
  firstNameErr?: string,
  lastName?: string,
  lastNameErr?: string,
  email?: string,
  emailErr?: string,
  date?: any
  dateErr?: string
  submitInfo?: string,
  avaibleDates?: any[],
  selectedDate?: any
}

export interface FormProps {
  match?: any,
  history?: any
}

export default class ApplicationForm extends React.Component<FormProps, FormState> {
  public happeningId = '';
  private applicationFormService: ApplicationFormService = new ApplicationFormService()
  constructor(props) {
    super(props);

    this.state = {
      date: '',
      title: '',
      firstName: '',
      lastName: '',
      email: '',
      submitInfo: '',
      avaibleDates: [],
      selectedDate: null
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleDateChange = this.handleDateChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);

    this.happeningId = this.props.match.params.id;
  }

  componentDidMount() {
    this.getHappening();
  }

  getHappening() {
    this.applicationFormService.getHappening(this.happeningId)
      .then((res) => {
        this.setState({
          title: res.title,
          avaibleDates: res.avaibleDates,
          selectedDate: moment(res.avaibleDates[0])
        })
      })
      .catch((error) => {
        this.setState({ submitInfo: 'There is problem with server connection' })
      });
  }

  handleChange = (e: React.FormEvent<EventTarget>): void => {
    let target = e.target as HTMLInputElement;
    this.setState({ [target.name]: target.value })
    if (this.state.submitInfo !== '') {
      this.setState({ submitInfo: '' })
    }
  }

  handleDateChange(date) {
    let dates = date.format("YYYY-MM-DD")
    this.setState({
      date: dates,
      selectedDate: date
    });
  }

  validate = () => {
    let state = {
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      email: this.state.email,
      date: this.state.date,
      avaibleDates: [...this.state.avaibleDates]
    }

    let validationRes = this.applicationFormService.validateForm(state);
    let isError = validationRes.isError
    let errors = validationRes.errors;

    this.setState({
      ...this.state,
      ...errors
    });

    return isError;
  };

  onSubmit = e => {
    e.preventDefault();
    const err = this.validate();
    if (!err) {
      let application = {
        happeningId: this.happeningId,
        firstName: this.state.firstName,
        lastName: this.state.lastName,
        email: this.state.email,
        date: this.state.date
      }
      this.applicationFormService.AddNewApplication(application)
        .then((response) => {
          this.setState({ submitInfo: response })
          if (response === 'Form correctly saved!') {
            setTimeout(() => {
              this.props.history.push("/")
            }, 1000)
          }
        })

    } else {
      this.setState({ submitInfo: '' })
    }
  };

  TextMaskCustom(props) {
    const { inputRef, ...other } = props;

    return (
      <MaskedInput
        {...other}
        ref={inputRef}
        mask={[/\d/, /\d/, /\d/, /\d/, '-', /\d/, /\d/, '-', /\d/, /\d/]}
        keepCharPositions={true}
        placeholder={'yyyy-mm-dd'}
      />
    );
  }

  render() {
    return (
      <div className="form-container">
        <form className='form' onSubmit={this.onSubmit}>
          <p className='title'>Sign up for {this.state.title}</p>

          <FormControl className="text-field" >
            <InputLabel >First name</InputLabel>
            <Input name="firstName" value={this.state.firstName} onChange={this.handleChange} />
            <FormHelperText className="field-error">{this.state.firstNameErr}</FormHelperText>
          </FormControl>

          <FormControl className="text-field" >
            <InputLabel >Last name</InputLabel>
            <Input name="lastName" value={this.state.lastName} onChange={this.handleChange} />
            <FormHelperText className="field-error">{this.state.lastNameErr}</FormHelperText>
          </FormControl>

          <FormControl className="text-field">
            <InputLabel >Email</InputLabel>
            <Input name="email" value={this.state.email} onChange={this.handleChange} />
            <FormHelperText className="field-error">{this.state.emailErr}</FormHelperText>
          </FormControl>

          <DatePicker
            selected={this.state.selectedDate}
            onChange={this.handleDateChange}
            includeDates={this.state.avaibleDates}
            customInput={
              <FormControl className="text-field"  >
                <InputLabel>Date</InputLabel>
                <Input name="date" value={this.state.date} onChange={this.handleChange} inputComponent={this.TextMaskCustom} />
                <FormHelperText className="field-error">{this.state.dateErr}</FormHelperText>
              </FormControl>
            }
          />

          <p className="submit-info">{this.state.submitInfo}</p>

          <Button variant="outlined" type='submit' className='submit-button'>
            Submit
          </Button>

        </form>
      </div>
    )
  }
}
