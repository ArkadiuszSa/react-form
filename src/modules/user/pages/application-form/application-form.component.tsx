import * as React from "react";
import "react-datepicker/dist/react-datepicker.css";
import DatePicker from "react-datepicker";
import Button from "@material-ui/core/Button";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import FormHelperText from "@material-ui/core/FormHelperText";
import FormControl from "@material-ui/core/FormControl";
import MaskedInput from "react-text-mask";

import "./application-form.scss";
import * as moment from "moment";

export interface FormProps {
  match: any;
  history: any;
  fetchHappening: Function;
  resetApplicationForm: Function;
  handleFormChange: Function;
  handleDateChange: any;
  submitApplicationForm: Function;
  title: string;
  firstName: string;
  firstNameErr: string;
  lastName: string;
  lastNameErr: string;
  email: string;
  emailErr: string;
  date: string;
  dateErr: string;
  avaibleDates: any[];
  selectedDate: any;
  submitInfo: string;
}

class ApplicationForm extends React.Component<FormProps, {}> {
  public happeningId: string = "id";
  public disabledInput: boolean = false;
  constructor(props) {
    super(props);
    this.happeningId = this.props.match.params.id;
  }
  componentWillMount() {
    this.props.resetApplicationForm();
    this.disabledInput = false;
  }

  componentWillUnmount() {
    this.props.resetApplicationForm();
  }

  componentDidMount() {
    this.props.fetchHappening(this.happeningId);
  }

  TextMaskCustom(props) {
    const { inputRef, ...other } = props;

    return (
      <MaskedInput
        {...other}
        ref={inputRef}
        mask={[/\d/, /\d/, /\d/, /\d/, "-", /\d/, /\d/, "-", /\d/, /\d/]}
        keepCharPositions={true}
        placeholder={"yyyy-mm-dd"}
      />
    );
  }

  render() {
    if (this.props.submitInfo === "Form correctly saved!") {
      this.disabledInput = true;
      setTimeout(() => {
        this.props.history.push("/");
      }, 2000);
    }
    if (this.props.title === "There is problem with server connection") {
      this.disabledInput = true;
      setTimeout(() => {
        this.props.history.push("/");
      }, 5000);
    }
    return (
      <div className="form-container">
        <form
          className="form"
          onSubmit={e =>
            this.props.submitApplicationForm(e, {
              happeningId: this.happeningId,
              firstName: this.props.firstName,
              lastName: this.props.lastName,
              email: this.props.email,
              date: this.props.date,
              avaibleDates: this.props.avaibleDates
            })
          }
        >
          <p className="title">{this.props.title}</p>

          <FormControl className="text-field">
            <InputLabel>First name</InputLabel>
            <Input
              disabled={this.disabledInput}
              name="firstName"
              value={this.props.firstName}
              onChange={e => this.props.handleFormChange(e)}
            />
            <FormHelperText className="field-error">
              {this.props.firstNameErr}
            </FormHelperText>
          </FormControl>

          <FormControl className="text-field">
            <InputLabel>Last name</InputLabel>
            <Input
              disabled={this.disabledInput}
              name="lastName"
              value={this.props.lastName}
              onChange={e => this.props.handleFormChange(e)}
            />
            <FormHelperText className="field-error">
              {this.props.lastNameErr}
            </FormHelperText>
          </FormControl>

          <FormControl className="text-field">
            <InputLabel>Email</InputLabel>
            <Input
              disabled={this.disabledInput}
              name="email"
              value={this.props.email}
              onChange={e => this.props.handleFormChange(e)}
            />
            <FormHelperText className="field-error">
              {this.props.emailErr}
            </FormHelperText>
          </FormControl>

          <DatePicker
            disabled={this.disabledInput}
            selected={this.props.selectedDate}
            onChange={this.props.handleDateChange}
            includeDates={this.props.avaibleDates}
            customInput={
              <FormControl className="text-field">
                <InputLabel>Date</InputLabel>
                <Input
                  disabled={this.disabledInput}
                  name="date"
                  value={this.props.date}
                  onChange={e => this.props.handleFormChange(e)}
                  inputComponent={this.TextMaskCustom}
                />
                <FormHelperText className="field-error">
                  {this.props.dateErr}
                </FormHelperText>
              </FormControl>
            }
          />

          <p className="submit-info">{this.props.submitInfo}</p>
          <Button
            disabled={this.disabledInput}
            variant="outlined"
            type="submit"
            className="submit-button"
          >
            Submit
          </Button>
        </form>
      </div>
    );
  }
}

export default ApplicationForm;
