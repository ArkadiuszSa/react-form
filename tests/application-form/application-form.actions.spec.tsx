import configureStore from "redux-mock-store";
import * as actions from "../../src/modules/user/pages/application-form/application-form.actions";

const mockStore = configureStore();
const store = mockStore();

describe("ApplicationForm-actions", () => {
  beforeEach(() => {
    store.clearActions();
  });

  describe("FETCH_APPLICATION_HAPPENING", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.fetchHappeningSucces("payload"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("FETCH_APPLICATION_HAPPENING", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.fetchHappeningFailure());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("RESET_APPLICATION_FORM", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.resetApplicationForm());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("HANDLE_FORM_CHANGE", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.handleFormChange("payload"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("HANDLE_DATE_CHANGE", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.handleDateChange("payload"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("VALIDATION_APPLICATION_FORM_FAILURE", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.validationApplicationFormFailure("payload"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("VALIDATION_APPLICATION_FORM_SUCCES", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.validationApplicationFormSucces());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("SUBMIT_APPLICATION_FORM", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.submitApplicationForm("payload"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("SUBMIT_APPLICATION_FORM_SUCCES", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.submitApplicationFormSucces());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("SUBMIT_APPLICATION_FORM_FAILURE", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.submitApplicationFormFailure("payload"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("RESET_APPLICATION_FORM_SUBMIT_INFO", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(actions.resetApplicationFormSubmitInfo());
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
