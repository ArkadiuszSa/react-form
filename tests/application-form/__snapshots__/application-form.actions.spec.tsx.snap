// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationForm-actions FETCH_APPLICATION_HAPPENING Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "payload",
    "type": "FETCH_APPLICATION_HAPPENING_SUCCES",
  },
]
`;

exports[`ApplicationForm-actions FETCH_APPLICATION_HAPPENING Dispatches the correct action and payload 2`] = `
Array [
  Object {
    "payload": "There is problem with server connection",
    "type": "FETCH_APPLICATION_HAPPENING_FAILURE",
  },
]
`;

exports[`ApplicationForm-actions HANDLE_DATE_CHANGE Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "payload",
    "type": "HANDLE_DATE_CHANGE",
  },
]
`;

exports[`ApplicationForm-actions HANDLE_FORM_CHANGE Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "payload",
    "type": "HANDLE_FORM_CHANGE",
  },
]
`;

exports[`ApplicationForm-actions RESET_APPLICATION_FORM Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "RESET_APPLICATION_FORM",
  },
]
`;

exports[`ApplicationForm-actions RESET_APPLICATION_FORM_SUBMIT_INFO Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "RESET_APPLICATION_FORM_SUBMIT_INFO",
  },
]
`;

exports[`ApplicationForm-actions SUBMIT_APPLICATION_FORM Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "payload",
    "type": "SUBMIT_APPLICATION_FORM",
  },
]
`;

exports[`ApplicationForm-actions SUBMIT_APPLICATION_FORM_FAILURE Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "payload",
    "type": "SUBMIT_APPLICATION_FORM_FAILURE",
  },
]
`;

exports[`ApplicationForm-actions SUBMIT_APPLICATION_FORM_SUCCES Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "Form correctly saved!",
    "type": "SUBMIT_APPLICATION_FORM_SUCCES",
  },
]
`;

exports[`ApplicationForm-actions VALIDATION_APPLICATION_FORM_FAILURE Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "payload",
    "type": "VALIDATION_APPLICATION_FORM_FAILURE",
  },
]
`;

exports[`ApplicationForm-actions VALIDATION_APPLICATION_FORM_SUCCES Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "VALIDATION_APPLICATION_FORM_SUCCES",
  },
]
`;
